src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    internal partial class ValueTrackingService
+    {
+        private class OperationCollector
+        {
+            public ValueTrackingProgressCollector ProgressCollector { get; }
+            public Solution Solution { get; }
+
+            public OperationCollector(ValueTrackingProgressCollector progressCollector, Solution solution)
+            {
+                ProgressCollector = progressCollector;
+                Solution = solution;
+            }
+
+            public Task VisitAsync(IOperation operation, CancellationToken cancellationToken)
+                => operation switch
+                {
+                    IObjectCreationOperation objectCreationOperation => VisitObjectCreationAsync(objectCreationOperation, cancellationToken),
+                    IInvocationOperation invocationOperation => VisitInvocationAsync(invocationOperation, cancellationToken),
+                    ILiteralOperation literalOperation => VisitLiteralAsync(literalOperation, cancellationToken),
+                    IReturnOperation returnOperation => VisitReturnAsync(returnOperation, cancellationToken),
+                    IArgumentOperation argumentOperation => ShouldTrackArgument(argumentOperation) ? VisitAsync(argumentOperation.Value, cancellationToken) : Task.CompletedTask,
+                    ILocalReferenceOperation or
+                        IParameterReferenceOperation or
+                        IFieldReferenceOperation or
+                        IPropertyReferenceOperation => VisitReferenceAsync(operation, cancellationToken),
+                    IAssignmentOperation assignmentOperation => VisitAssignmentOperationAsync(assignmentOperation, cancellationToken),
+
+                    // Default to reporting if there is symbol information available
+                    _ => VisitDefaultAsync(operation, cancellationToken)
+                };
+
+            private async Task VisitDefaultAsync(IOperation operation, CancellationToken cancellationToken)
+            {
+                // If the operation has children, always visit the children instead of the root 
+                // operation. They are the interesting bits for ValueTracking
```

Comment:
this commen could use more info to explain why this is.

Reply:
Changing this to explicitly handle operations that we want to visit children of in VisitAsync. This did a good job of catching all the cases, but being explicit would make the code easier to upkeep and read.